docs/guide/tngfue-installation.md

```diff
@@ -0,0 +1,210 @@
+(same as the AP setting part)
+- `identity:` identity information for this ue 
+(set whatever you want)
+- `password:` wifi/ap password
+(same as the AP setting part)
+
+```
+ctrl_interface=udp
+update_config=1
+network={
+    ssid="<YOUR_AP_NAME>"
+    key_mgmt=WPA-EAP
+    eap=VENDOR-TEST IKEV2
+    identity="<YOUR_IDENTITY_INFO>"
+    password="<YOUR_PASSWORD>"
+}
+```
+
+E.g.
+```
+ctrl_interface=udp
+update_config=1
+network={
+    ssid="free5gc-ap"
+    key_mgmt=WPA-EAP
+    eap=VENDOR-TEST IKEV2
+    identity="tngfue"
+    password="free5gctngf"
+}
+```
+
+#### Security Parameters
+In ```tngfue/wpa_supplicant/sec.conf```:
+
+- `wifiifname:` your wifi interface name 
+(you can find this by typing `ifconfig` in terminal)
+
+You can find all the parameters listed below on the Web Console subscriber page.
+
+- `K:` 
+- `imsi_identity:` SUPI(IMSI)
+- `MSIN:` SUPI without the PLMN part
+- `SQN:`
+- `AMF:` Authentication Management Field
+- `OPC:` Operator Code Value
+
+Format: 
+```
+wifiifname: <YOUR_WIFI_INTERFACE_NAME>
+K:<UE_K_IN_WEBCONSOLE>
+imsi_identity:<UE_SUPI_IN_WEBCONSOLE>
+nai_username:type0.rid61695.schid0.userid<MSIN>
+SQN:<UE_SQN_IN_WEBCONSOLE>
+AMF:<AMF_IN_WEBCONSOLE>
+OPC:<UE_OPCV_IN_WEBCONSOLE>
+```
+
+E.g.
+```
+wifiifname:wlp3s0
+K:8baf473f2f8fd09487cccbd7097c6862
+imsi_identity:208930000000007
+nai_username:type0.rid61695.schid0.userid000000000007
+SQN:16f3b3f70fe1
+AMF:8000
+OPC:8e27b6af0e692e750f32667a3b14605d
+```
+
+#### Build binary
+```
+cd ~/tngfue/wpa_supplicant/
+make
+```
+
+**Trouble shooting: Missing packages when building binary**
+
+Update Packages source before installing packages!
+```
+sudo apt-get update
+```
+
+openssl/ssl.h: No such file or directory
+```
+sudo apt-get install libssl-dev
+```
+dbus/dbus.h: No such file or directory
+```
+sudo apt-get install libdbus-1-dev
+```
+netlink/netlink.h: No such file or directory
+```
+sudo apt-get install libnl-3-dev
+```
+/usr/bin/ld: cannot find -lnl-genl-3
+```
+sudo apt-get install libnl-genl-3-dev
+```
+/usr/bin/ld: cannot find -lnl-route-3
+```
+sudo apt-get install libnl-route-3-dev
+```
+
+## 5. Testing
+### A. Run free5GC with tngf
+```
+cd ~/free5gc
+./run.sh -tngf
+```
+### B. Run tngfue
+#### Make sure xfrm and greTun0 is not exist
+> If you don't have a successful connection with TNGF before, you can just skip this step.
+
+Delete two interfaces
+```
+sudo ip l del greTun0
+sudo ip l del xfrm-1
+```
+#### Set wifi interface IP to the same LAN with free5GC
+```
+ip a add 192.168.1.202/24 brd 192.168.1.255 dev <YOUR_WIFI_INTERFACE_NAME>
+ip r add default via 192.168.1.202 dev <YOUR_WIFI_INTERFACE_NAME>
+```
+#### Run
+```
+cd ~/tngfue/wpa_supplicant
+sudo ./wpa_supplicant -c ../wpa_supplicant.conf -i <YOUR_WIFI_INTERFACE_NAME> -dd
+```
+
+When the session connects successfully, you will see these five interfaces established. 
+![image](./TNGFUE/interfaces.png)
+
+
+### C. Transfer data over the greTun0 interface
+
+```
+ping -I greTun0 8.8.8.8
+```
```

Comment:
At the below, provide some screenshots that indicate the TNGFUE is successful start and run.